base/sort.jl

```diff
@@ -1162,25 +1183,40 @@ julia> v[p]
  1
  2
  3
+
+julia> A = [8 7; 5 6]; p = zeros(Int,2, 2);
+
+julia> sortperm!(p, A; dims=1); p
+2×2 Matrix{Int64}:
+ 2  4
+ 1  3
+
+julia> sortperm!(p, A; dims=2); p
+2×2 Matrix{Int64}:
+ 3  1
+ 2  4
 ```
 """
-function sortperm!(x::AbstractVector{T}, v::AbstractVector;
+function sortperm!(ix::AbstractArray{<:Integer}, A::AbstractArray;
```

Comment:
```suggestion
function sortperm!(ix::AbstractArray{T}, A::AbstractArray;
```
Workspace and index vectors must have the same eltype.